### ISSUE TITLE
Postgres-operator severed from cluster by changing `postgres.*.credentials` secret

### ISSUE THREAD

**Issue by kupson** [zalando/postgres-operator#524]
Hi,

Changing `postgres` user password by modifying Kubernetes `postgres.*.credentials` Secret causes error in postgres-operator:

```
time="2019-03-27T18:51:01Z" level=warning msg="error while syncing cluster state: could not sync roles: could not init db connection: could not init db connection: pq: password authentication failed for user \"postgres\"" cluster-name=postgres-operator-pg-test/infra-postgres-operator-test-patroni pkg=cluster
time="2019-03-27T18:51:01Z" level=error msg="could not sync cluster: could not sync roles: could not init db connection: could not init db connection: pq: password authentication failed for user \"postgres\"" cluster-name=postgres-operator-pg-test/infra-postgres-operator-test-patroni pkg=controller worker=0
```

The postgres-operator cannot connect to the Postgres because it's trying to use the new password (from Secret) and fails.

Possible solutions:
- port-forward so the connection will be authenticated by using trusted IP address in `pg_hba.conf`
- extend Patroni REST API to be able to change superuser password after initialisation
- exec into master spilo Pod and run psql from there (yuk!)

**Comment by FxKu** [comment]
Alternatively, you can delete the operator Pod. This should trigger a syncing of the secrets with the DB.

**Comment by kupson** [comment]
Yeah, the erroneous state can easily be fixed by operator but it would be better to let postgres-operator handle it internally in a correct way. Doing exit or panic in the process in such case it not ideal - why one affected database would cause restart of the whole operator possibly affecting multiple users?

**Comment by FxKu** [comment]
I agree that it could be handled in a nicer way, but why would a restart of the operator pod affect multiple users? It gets re-spawned immediately. The postgres clusters stay in place as they were.

**Comment by kupson** [comment]
I didn't tested it but what would happen if someone (or some runaway script) would change password every 1s? Postgres operator would probably do some work before crashing but that would slow things down to a crawl.

Do the other reasons (other than bad password) for failed sync roles operations would cause `panic()` too?

To me it looks like the current behaviour is safer than panic() but could be improved.

Restarting operator Pod as a manual action is OK. Crashes and restarts might let to other unexpected behaviour and worsen the user experience.
